### PR TITLE
Add KDE Connect [Need Help with Livecheck regex]

### PR DESCRIPTION
**Pull request by CodeWithShreyans**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### Need help with Livecheck regex

### CODE REVIEW

**Comment by DestroyerXyz** [comment]
I can't find the app id for exit in the uninstall stanza

**Comment by DestroyerXyz** [comment]
I messed something up on my machine, installing and uninstalling too many times 😛

File: Casks/kdeconnect.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  sha256 "8775967bf73470c3064724235608e982208bdee8f8d719ecd03bb9313205cf5c"
+
+  url "https://binary-factory.kde.org/job/kdeconnect-kde_Release_macos/lastSuccessfulBuild/artifact/kdeconnect-kde-#{version}-macos-64-clang.dmg"
+  name "KDE Connect"
+  desc "Enabling communication between all your devices"
+  homepage "https://kdeconnect.kde.org/"
+
+  livecheck do
+    url "https://binary-factory.kde.org/job/kdeconnect-kde_Release_macos/lastSuccessfulBuild/artifact/"
+    regex(/href=.*?kdeconnect-kde[._-]v?(\d+(?:[.-]\d+)+)-macos-64-clang.dmg/i)
```
Comment (neutric) [inline Casks/kdeconnect.rb:12]: `clang.dmg` => `clang\.dmg`

File: Casks/kdeconnect.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name "KDE Connect"
+  desc "Enabling communication between all your devices"
+  homepage "https://kdeconnect.kde.org/"
+
+  livecheck do
+    url "https://binary-factory.kde.org/job/kdeconnect-kde_Release_macos/lastSuccessfulBuild/artifact/"
+    regex(/href=.*?kdeconnect-kde[._-]v?(\d+(?:[.-]\d+)+)-macos-64-clang\.dmg/i)
+  end
+
+  app "kdeconnect-indicator.app", target: "KDE Connect.app"
```
Comment (miccal) [inline Casks/kdeconnect.rb:15]: Why the name change? We generally do not do this and follow the naming from upstream unless there is a valid reason.

Comment (DestroyerXyz) [reply Casks/kdeconnect.rb:15]: I think the naming scheme is a little confusing and changing it will make it easier to find in the Applications folder.

Comment (SMillerDev) [reply Casks/kdeconnect.rb:15]: You can probably ask the KDE project to just rename it then.

Comment (hybras) [reply Casks/kdeconnect.rb:15]: The app's [building docs](https://community.kde.org/KDEConnect/Build_MacOS), [download instructions](https://kdeconnect.kde.org/download.html), [repo name](https://invent.kde.org/network/kdeconnect-kde) are inconsistent in whether the dash is included and whether 'indicator' is included. 

The source code and built artifacts seems to unequivocally use indicator.

**Comment by hybras** [comment]
As much as I want this to be merged, I'm not sure if Connect is stable software? The PR Author found release builds, but the [official download page](https://kdeconnect.kde.org/download.html) directs to the nightly builds. The [repo](https://invent.kde.org/network/kdeconnect-kde) declares macos as not officially supported.

**Comment by DestroyerXyz** [comment]
The nightly builds have been abandoned. Instead, release builds are built nightly.

**Comment by miccal** [comment]
Thank you @DestroyerXyz, but I feel that this is not suitable for addition to Cask at this stage.

Please consider hosting this [in your own Tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) and if it gets more stable and better supported we would then be happy to consider adding this to `homebrew-cask`.

**Comment by hybras** [comment]
@DestroyerXyz I made a [cask](https://github.com/hybras/homebrew-tap/blob/main/Casks/kdeconnect.rb) in my tap based on your work. You can discuss it at the [pr](https://github.com/hybras/homebrew-tap/pull/1). I changed the url to download from the appropriate job, not from the latest sucessfull (latest is still used in livecheck).

**Pull request closed without merging**